distributed/scheduler.py

```diff
@@ -1948,11 +1829,7 @@ def decide_worker(self, ts: TaskState) -> WorkerState:  # -> WorkerState | None
             else:  # dumb but fast in large case
                 ws = wp_vals[self.n_tasks % n_workers]
 
-        if self.validate:
-            assert ws is None or isinstance(ws, WorkerState), (
-                type(ws),
-                ws,
-            )
+        if self.validate and ws is not None:
             assert ws.address in self.workers
```

Comment:
!!! The top-level function ``decide_workers`` NEVER returns None in the unit tests !!!

Reply:
https://github.com/dask/distributed/issues/6254